framework/src/main/java/org/checkerframework/framework/source/SourceChecker.java

```diff
@@ -1790,27 +1775,27 @@ protected void warnUnneededSuppressions() {
 
         Set<Element> elementsSuppress = new HashSet<>(this.elementsWithSuppressedWarnings);
         this.elementsWithSuppressedWarnings.clear();
-        Set<String> checkerKeys = new HashSet<>(getSuppressWarningsKeys());
+        Set<String> prefixes = new HashSet<>(getSuppressWarningsPrefixes());
         Set<String> errorKeys = new HashSet<>(messagesProperties.stringPropertyNames());
-        warnUnneededSuppressions(elementsSuppress, checkerKeys, errorKeys);
+        warnUnneededSuppressions(elementsSuppress, prefixes, errorKeys);
         getVisitor().treesWithSuppressWarnings.clear();
     }
 
     /**
      * Issues a warning about any {@code @SuppressWarnings} that isn't used by this checker, but
-     * contains a key that would suppress a warning from this checker.
+     * contains a string that would suppress a warning from this checker.
      *
      * @param elementsSuppress elements with a {@code @SuppressWarnings} that actually suppressed a
      *     warning
-     * @param checkerKeys suppress warning keys that suppress any warning from this checker
-     * @param errorKeys error keys that can be issued by this checker
+     * @param prefixes Suppresswarnings prefixes that suppress any warning from this checker
+     * @param allErrorKeys all error keys that can be issued by this checker
      */
     protected void warnUnneededSuppressions(
```

Comment:
The implementations of
  `SourceChecker.shouldSuppressWarningStringSuppressKey`
  `SourceChecker.warnUnneededSuppressions`
are very similar.
Either abstract out common logic, or eliminate one of these two methods.
(Note that there are some undesirable differences, such as the logic for permitting a user-written messagekey to be a substring of a full Checker Framework messagekey.)